### PR TITLE
Port FluxRegister::ClearInternalBorders to GPU

### PR DESCRIPTION
**Pull request by WeiqunZhang**
## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] are described in the proposed changes to the AMReX documentation, if appropriate

### CODE REVIEW

**Comment by zingale** [comment]
This works with Castro + https://github.com/AMReX-Astro/Castro/pull/1632

**Review by maximumcats — APPROVED** [review]

File: Src/AmrCore/AMReX_FluxRegister.cpp [hunk 1]
```diff
@@ -616,22 +625,40 @@ FluxRegister::ClearInternalBorders (const Geometry& geom)
...
+                            }
                         }
                     }
                 }
             }
         }
     }
+
+#ifdef AMREX_USE_GPU
+    // There is Gpu::synchronize in Parallefor below internally.
```
Comment (bl-ue) [inline Src/AmrCore/AMReX_FluxRegister.cpp:655]: Typo? `Parallefor` → `ParallelFor`
```diff
-Parallefor
+ParallelFor
```

Comment (WeiqunZhang) [reply Src/AmrCore/AMReX_FluxRegister.cpp:655]: Yes that's a typo.

Comment (bl-ue) [reply Src/AmrCore/AMReX_FluxRegister.cpp:655]: I can open a PR to fix it, if you'd like.

Comment (WeiqunZhang) [reply Src/AmrCore/AMReX_FluxRegister.cpp:655]: Sure.  Thanks!

Comment (bl-ue) [reply Src/AmrCore/AMReX_FluxRegister.cpp:655]: @WeiqunZhang do you mind if I fix all of the typos across the code base (using [codespell](https://github.com/codespell-project/codespell))? Things like "virtural", "retruned", "apoints", "funciton", "sequencially", "untill", "occurence", etc.

Comment (WeiqunZhang) [reply Src/AmrCore/AMReX_FluxRegister.cpp:655]: Fine with me.

Comment (bl-ue) [reply Src/AmrCore/AMReX_FluxRegister.cpp:655]: Okay, I'll do that.  To give you a hint about the volume, there's about ~437 :smile:

Still ok with you?

Comment (WeiqunZhang) [reply Src/AmrCore/AMReX_FluxRegister.cpp:655]: yes

**Pull request merged**